src/Type/WPInterfaceTrait.php

```diff
@@ -227,20 +252,65 @@ private function get_fields_from_implemented_interfaces( TypeRegistry $registry
 				continue;
 			}
 
-			$interface_config_fields = $interface_type->getFields();
+			// Get all interfaces in the inheritance chain
+			$all_interfaces = $this->get_all_interfaces( $interface_type, $registry );
```

Comment:
but `::get_type()` gives us the _constructed_ type with all the resolved fields (including from the interface) because of our `get_fields()` this seems like weird recursion or am I missing something?